### PR TITLE
Upload stage timing from perf pipeline to kusto

### PR DESCRIPTION
**Pull request by NicholasCouri**
## Description

Today, the pipelines get broken frequently and we do not have a way to generate alerts whenever it happens. As a proof of concept, this change adds a stage to the performance benchmark that makes a simple REST call to the timeline API from dev azure to request information about the current build id. With that information in hand, we upload to ARIA and, as a result, we will be able to create monitors and alerts based on the stage results as well as find out avg duration of each stage. 
The idea is to do something similar to other pipelines as the need arises.

@alexvy86  I believe there is a way to dynamically get the name of all previous stages and use them to make sure the last one to be run is the one we've created. For now, I kept it hardcoded but ideally, we should move in that direction. Thoughts? 
Also, notice I'm only using the stages and not the tasks (there are TOO many tasks and for now I will not upload them).

### CODE REVIEW

File: tools/pipelines/test-perf-benchmarks.yml [hunk 1]
```diff
@@ -611,3 +611,38 @@ stages:
...
+          - task: Bash@3
+            displayName: Retrieve buildId results
+            inputs:
+              targetType: 'inline'
+              workingDirectory: $(toolAbsolutePath)
+              script: |
+                echo "creating output folder"
+                mkdir -p ${{ variables.testWorkspace }}/timingOutput
+                echo "Executing curl command ..."
+                echo  'curl -u ":$(System.AccessToken)" "https://dev.azure.com/fluidframework/internal/_apis/build/builds/$(Build.BuildId)/timeline"'
```
Comment (anthony-murphy) [inline tools/pipelines/test-perf-benchmarks.yml:639]: don't log credentials

Comment (NicholasCouri) [reply tools/pipelines/test-perf-benchmarks.yml:639]: they show up as ***  , so I assume we should be fine.
Ex: 
curl -u ":***" "https://dev.azure.com/***framework/internal/_apis/build/builds/162093/timeline"
https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_apis/build/builds/162093/logs/150

**Review by tyler-cai-microsoft — APPROVED** [review]
LGTM :shipit:

**Comment by NicholasCouri** [comment]
I will merge but please let me know if there are any concerns and we can address them.

**Pull request merged**